### DISCUSSION TITLE
Transactions, are they possible?

### DISCUSSION THREAD

**Discussion by mike-kolback** [Gremlinq/ExRam.Gremlinq#225] [Q&A] [answered]
Hi All,

My team and I are currently using GremlinQ within a couple of our services.  One issue we are beginning to see is a lack of consistency with some of our more complex queries.  In our current architecture, we are building out some pretty complex subgraphs all at once.  Now because of this complexity, were starting to run into issues with idempotency.  Some calls perform as expected, and others tend to see errors or complete successfully with missing vertices / edges.  I know that Tinkerpop supports the concept of transactions, but I'm not sure if that's something that extends to Gremlin.NET.  My question is, is it possible to support a more transactional behavior so as to ensure idempotency.

And if this is outside of the scope of GremlinQ (It very well may be) I apologize.  Just looking for some insights.

Thanks!

**Comment by danielcweber** [comment, marked as answer]
Transactions is, at least for the cloud providers that I'm aware of, something that is either not supported at all or supported implicitly, i.e. one transaction per query (Neptune). As such, there is no explicit support for transactions in Gremlinq because it's more or less pointless anyway, at least as things are currently.

**Reply by mike-kolback** [reply]
Thats what I figured.  Thanks for the confirmation!
